services/s3/src/it/java/software/amazon/awssdk/services/s3/S3IntegrationTestBase.java

```diff
@@ -73,8 +73,18 @@ protected static void createBucket(String bucketName) {
                                                                         .build())
                                        .build());
         } catch (S3Exception e) {
+            System.err.println("Error attempting to create bucket: " + bucketName);
             if (e.getErrorCode().equals("BucketAlreadyOwnedByYou")) {
                 System.err.printf("%s bucket already exists, likely leaked by a previous run\n", bucketName);
+            } else if (e.getErrorCode().equals("TooManyBuckets")) {
+                System.err.println("Printing all buckets for debug:");
+                s3.listBuckets().buckets().forEach(System.err::println);
+                if (retryCount < 2) {
```

Comment:
Are we expecting that some tests running in parallel might finish and clean up resources? Is that why we retry for "TooManyBuckets" case?

Reply:
that's my theory. we keep getting that exception but then by the time i go and look at the console we're back to 30 buckets.